src/main/scala/chisel3/util/experimental/LoadMemoryTransform.scala

```diff
@@ -0,0 +1,231 @@
+// See LICENSE for license details.
+
+package chisel3.util.experimental
+
+import chisel3._
+import chisel3.experimental.annotate
+import chisel3.InstanceId
+import chisel3.experimental.{ChiselAnnotation, RunFirrtlTransform}
+import firrtl.annotations.{MemoryLoadFileType, _}
+import firrtl.ir.{Module => _, _}
+import firrtl.transforms.BlackBoxInlineAnno
+import firrtl.Mappers._
+import firrtl.{AnnotationSeq, CircuitForm, CircuitState, EmitCircuitAnnotation, LowForm, Transform, VerilogEmitter}
+
+import scala.collection.mutable
+
+/** Use this annotation to load a memory from a text file by using verilator and verilog's $readmemh or $readmemb
+  * The treadle backend can also recognize this annotation and load memory at run-time.
+  *
+  * This annotation triggers the [[LoadMemoryTransform]] which will take add the verilog directive to
+  * the relevant module by using the creating separate modules that are bound to the modules containing
+  * the memories to be loaded.
+  *
+  * ==Example module==
```

Comment:
Great documentation. I'd move the body onto the `loadMemoryFromFile` object.

Reply:
Moved doc code to loadMemoryFromFile which is how annotation is created